compiler/vm/vmgen.nim

```diff
@@ -1337,22 +1337,33 @@ proc genBinaryABC(c: var TCtx; n: CgNode; dest: var TDest; opc: TOpcode) =
   c.freeTemp(tmp)
   c.freeTemp(tmp2)
 
-proc genNarrow(c: var TCtx; n: CgNode; dest: TDest) =
-  let t = skipTypes(n.typ, abstractVar-{tyTypeDesc})
-  # uint is uint64 in the VM, we we only need to mask the result for
-  # other unsigned types:
-  if t.kind in {tyUInt8..tyUInt32} or (t.kind == tyUInt and t.size < 8):
-    c.gABC(n, opcNarrowU, dest, TRegister(t.size*8))
-  elif t.kind in {tyInt8..tyInt32} or (t.kind == tyInt and t.size < 8):
-    c.gABC(n, opcNarrowS, dest, TRegister(t.size*8))
+proc genNarrow(c: var TCtx; n: CgNode; dest: TRegister; sNarrow = opcNarrowS) =
```

Comment:
```suggestion
proc genNarrow(c: var TCtx; info: TLineInfo; dest: TRegister; sNarrow = opcNarrowS) =
```

This could be a line info, instead of `n: CgNode`, in which case `genNarrowU` and `genNarrowUnsigned` could be simplified out or become inline `proc`s which specify the opcode and possibly carry some minimal logic.

Reply:
Only `genNarrowU` can be simplified away this way; `genNarrowUnsigned` deviates from the other narrow procedures in that it generates nothing for signed types.

I'll replace `genNarrowU`.

Reply:
I'm not sure how I feel about it, as it could be too clever, but `ocpRet` could potentially be used as a sentinel value to indicate "just return the value if signed"?

It's probably not a great idea.

Reply:
> but `ocpRet` could potentially be used as a sentinel value to indicate "just return the value if signed"?

Yep, that would work, although I think I'd rather leave `genNarrowUnsigned` be for the moment.